snuba/clickhouse/columns.py

```diff
@@ -190,6 +191,36 @@ def for_schema(self) -> str:
         return "{} DEFAULT {}".format(self.inner_type.for_schema(), self.default)
 
 
+class ReadOnly(ColumnType):
+    def __init__(self, inner_type: ColumnType) -> None:
+        self.inner_type = inner_type
+
+    def __repr__(self) -> str:
+        return "ReadOnly({})".format(repr(self.inner_type))
+
+    def get_raw(self) -> ColumnType:
+        return self.inner_type.get_raw()
+
+
+class Nullable(ColumnType):
```

Comment:
Can we have a `Nullable(Readonly(...))` ?  If yes I think we may want to represent these modifiers as properties of a type more than as types themselves. That would make it easier to unpack them in places like the validators.

Reply:
No you can't have `Nullable(ReadOnly(())` in the same way that you can't have `Nullable(Materialized())` - ReadOnly right now is a drop in replacement for Materialized. I agree it may be more usable to have it be a property of a type though.

Reply:
Are you planning to make any change in this direction (the properties) ?

Reply:
Not immediately, for now it's just an idle thought. We do need to keep things roughly compatible with the old system until it's removed at very least, which is the main reason ReadOnly is implemented in this way.